qubesadmin/tools/qvm_ls.py

```diff
@@ -710,16 +782,109 @@ def main(args=None, app=None):
             vm for vm in args.app.domains if vm.name not in args.exclude
         ]
 
+    if args.klass:
+        # filter only VMs to specific class(es)
+        domains = [d for d in domains if d.klass in args.klass]
+
+    if args.label:
+        # filter only VMs with specific label(s)
+        domains_labeled = []
+        spinner.show('Filtering based on labels...')
+        for dom in domains:
+            if dom.label.name in args.label:
+                domains_labeled.append(dom)
```

Comment:
> Was this for the spinner updates?

Yes. Exactly. If qvm-ls is run outside of dom0, the policy checks comes into play and it could become slow. And it could make it slow. I  considered moving it to Table class. But there might be cases that a field is required for filtering but not in output.